fetcher/src/job/job.utils.ts

```diff
@@ -92,23 +117,55 @@ function validateAdapter(adapter): IAdapter {
   }
 }
 
+function selectProxy(url: string, proxies) {
+  const source = url.split('/')[2]
+  const proxySize = proxies.length
+
+  if (!source) {
+    throw new FetcherError(FetcherErrorCode.InvalidUrl)
+  }
+  if (proxySize == 0) {
```

Comment:
I think it would be better to operate even though there are no proxies setup, otherwise we always have to use proxies even for the simplest use cases.

Reply:
If we follow my suggestion with `type Proxy` and check for `undefined` before accessing `host` and `port`, we don't have to raise this error, but just return undefined proxy.